DatadogSessionReplay/Sources/Recorder/TouchSnapshotProducer/TouchSnapshot/TouchIdentifierGenerator.swift

```diff
@@ -74,11 +74,17 @@ internal final class TouchIdentifierGenerator {
 // MARK: - UIView tagging
 
 fileprivate var associatedTouchIdentifierKey: UInt8 = 1
+fileprivate var associatedTouchPrivacyOverrideKey: UInt8 = 2
 
-private extension UITouch {
+internal extension UITouch {
     var identifier: TouchIdentifier? {
         set { objc_setAssociatedObject(self, &associatedTouchIdentifierKey, newValue, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC) }
         get { objc_getAssociatedObject(self, &associatedTouchIdentifierKey) as? TouchIdentifier }
     }
+
+    var touchPrivacyOverride: TouchPrivacyLevel? {
+        set { objc_setAssociatedObject(self, &associatedTouchPrivacyOverrideKey, newValue, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC) }
+        get { objc_getAssociatedObject(self, &associatedTouchPrivacyOverrideKey) as? TouchPrivacyLevel }
+    }
 }
```

Comment:
/suggestion I think it's a good opportunity to move this to a `DatadogExtended` type.

```suggestion
private var associatedTouchIdentifierKey: UInt8 = 1
private var associatedTouchPrivacyOverrideKey: UInt8 = 2

extension UITouch: DatadogExtended { }

extension DatadogExtension where ExtendedType: UITouch {
    var identifier: TouchIdentifier? {
        set { objc_setAssociatedObject(type, &associatedTouchIdentifierKey, newValue, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC) }
        get { objc_getAssociatedObject(type, &associatedTouchIdentifierKey) as? TouchIdentifier }
    }

    var touchPrivacyOverride: TouchPrivacyLevel? {
        set { objc_setAssociatedObject(type, &associatedTouchPrivacyOverrideKey, newValue, objc_AssociationPolicy.OBJC_ASSOCIATION_RETAIN_NONATOMIC) }
        get { objc_getAssociatedObject(type, &associatedTouchPrivacyOverrideKey) as? TouchPrivacyLevel }
    }
}
```

Reply:
Fair point!

I moved these properties to `DatadogExtended`, as you suggested. However, since `UITouch` is immutable, we need to create a mutable copy of the extension when assigning the properties.

Reply:
As discussed, I reverted this change.